### DISCUSSION TITLE
Blog + Federation

### DISCUSSION THREAD

**Discussion by nigini** [tinyBlogPub/microblog.pub#4] [Ideas]
As a user, I think that the Blog part of it is too intertwined with the Federation part of it.
What if I want to write something, maybe a draft, maybe a longer post that I don't want to send out right to the Fediverse?
At least from the perspective of the UI (I am still examining the backend code), there is no blogging without federation. I would love to have that.

What if, from the ground up, we rethink this software as a federatable, but more importantly, hyper-personal place? The minimum I'd suggest changing it would be to give control to the owner, what do you want to do with this post: draft, post on the website only, send it out to followers, etc?

**Comment by 3b0b** [comment]
I'm not sure either of those _necessarily_ requires "ground up" rethinking. I believe an actual draft mode was planned, though I'm not sure exactly how; I haven't gotten around to it yet - partly because I found how the ability to manage "pages" worked instead, and for my purposes that's almost as good - but I think I had a pretty solid idea of how to implement non-federated posts...

Wait.

No, on second thought, I'm not sure I did. I think what I had was an idea for how to keep a post from being _sent_ to any other instances. But what I wanted at the time was mostly a way to post something publicly without shoving it into my followers' timelines. I suspect if I did it the way I had in mind, it would technically still be a Fediverse post and if someone actually searched for it explicitly from their instance, or did the "interact from your instance" song and dance from looking at it on mine, it would still function as one otherwise. Which...I think for my purposes is actually a good thing, but might not be what you have in mind or what other people might want.

Setting aside the "whether to federate" part, I think an implementation of _the ability to edit_, including correct handling of sending updates in both contexts, might be able to substitute. At one point I tried using a mentioned-actors-only post that _didn't mention anyone_ as a draft mode, but to draft formatting without an edit mode (and without digging it out of the database on the server to work on it again), I would have needed to put it in a code block...and therefore not use any code blocks...plus it just seemed more of a hassle than it was worth, so I went back to using local .md files for my drafts.

Sorry, most of this turned into a not-very useful ramble.

**Reply by nigini** [reply]
Well, it is good to know that someone agrees with me that some Blogging UX improvements would be good. I especially want to be able to publish pages or longer posts to have a URL to them and then create a short federated post that points to that larger post. I think this is better aligned with the Fediverse. 

In summary:

* R1: I agree with the need for the **ability to edit a post**. Even more because it is part of the ActivityPub protocol.
* R2: Ability to **choose to federate posts**. We should likely have a *default* configuration for different types of posts. 
* R3 (?): Do we already **have an RSS feed**? That would be a much better way to distribute longer posts, right?

**Reply by chri2** [reply]
Doesn't that work, yet? Writing an **article** just does that exactly when I look at my microblog.pub following it from a mastodon account: There's the headline and an URL leading to the article, e.g.: https://chrichri.ween.de/articles/1fb1a0f/creating-a-team-account-for-the-fediverse .

The only thing I'd miss is editing an existing article with all that is needed to federate it (not sure about it, but I guess a `delete` for the former version and posting the new version).

For writing a draft there is no function in microblog.pub it seems and there has been an open issue on sr.ht around. I solved the problem by using either `retext` to draft the article offline or `hedgedoc` to draft online.

I'd like to see a draft function, too, but there are already good alternatives (especially hedgedoc, etherpad, cryptpad) to make this less important for me.

**Reply by nigini** [reply]
I agree that the edit functionality is a priority: I created issue #11.

I need to double-check the **article** posting experience. The described visualization is a pretty good start (but I believe it is dependent on the client.) My requirement, though, is to completely separate the federation from the post creation. In that case, I can personalize how I want to introduce my article, like by writing a smaller post that is federated AND has the link to the article.

**Reply by 3b0b** [reply]
If that's what you're after, then it may be that the entire trappings of ActivityPub aren't really something you want, either; another alternative you might find worth considering using the support for arbitrary _pages_...perhaps with one of them being a table of contents, so that people could _find_ those pages without having to scroll through your notes...and then write the notes that you describe linking to them.

I don't write notes introducing any of my pages, so far, because...well, I just haven't had any I'd want to do that with.

However, this does give up the editor; and in the absence of additional effort, it requires you to write html instead of markdown. Also, I _think_ if you merely prevent the active federation but still make the articles ActivityPub posts, then (to the potential confusion of your followers) people would still be able to search explicitly for your articles (or possibly for you) from their instances and read your articles there.

I don't necessarily recommend either approach over the other; your description of what you want, which is also a thing I used to want (and still would kind of like to have; I just don't want it enough to work on it myself anymore) prompts me to contemplate both how what you describe would ultimately function, and what else might achieve similar ends.

**Reply by nigini** [reply]
@3b0b Good to know someone other then me would appreciate the option to not federate some content.
The other part of what we have discussed here (editing content) seems to have some exciting improvements here: #11
